store/postgres/src/transaction_receipt.rs

```diff
@@ -75,32 +29,44 @@ select
     ethereum_hex_to_bytea(receipt ->> 'gasUsed') as gas_used,
     ethereum_hex_to_bytea(receipt ->> 'status') as status
 from (
-    select jsonb_array_elements(data -> 'transaction_receipts') as receipt
-    from "#,
-        );
-        out.push_sql(&self.blocks_table_name);
-        out.push_sql(" where hash = ");
-        out.push_bind_param::<Binary, _>(&self.block_hash)?;
-        out.push_sql(") as temp;");
-        Ok(())
-    }
-}
+    select
+        jsonb_array_elements(data -> 'transaction_receipts') as receipt
+    from
+        {blocks_table}
+    where hash = $1) as temp;
+",
+        blocks_table = blocks_table_name
+    ));
 
-impl<'a> Query for TransactionReceiptQuery<'a> {
-    type SqlType = (
-        Binary,
-        Binary,
-        Nullable<Binary>,
-        Nullable<Binary>,
-        Nullable<Binary>,
-        Nullable<Binary>,
-    );
-}
+    let query_results: Result<Vec<RawTransactionReceipt>, diesel::result::Error> = {
+        // The `hash` column has different types between the `public.ethereum_blocks` and the
+        // `chain*.blocks` tables, so we must check which one is being queried to bind the
+        // `block_hash` parameter to the correct type
+        if blocks_table_name == ETHEREUM_BLOCKS_TABLE_NAME {
```

Comment:
If this whole function was a method on `Storage` you could make the decision based on what `Storage` variant you have rather than relying on specific table names (see e.g. how `Storage.ancestor_block` works)